WangscapeTest/TestCartesianPower.cpp

```diff
@@ -0,0 +1,199 @@
+#include <gtest/gtest.h>
+
+#include <vector>
+#include <algorithm>
+
+#include <tilegen/CartesianPower.h>
+
+class TestCartesianPower : public ::testing::Test
+{
+protected:
+    typedef std::vector<int> Container;
+    typedef std::vector<int>::const_iterator Iterator;
+
+    Container base{4, 5, 6};
+    const tilegen::CartesianPower<Iterator> cp{base, 4};
+};
+
+TEST_F(TestCartesianPower, TestCartesianPowerSize)
+{
+    EXPECT_EQ(81, cp.size()) <<
+        "CartesianProduct has incorrect size";
+}
+
+TEST_F(TestCartesianPower, TestCartesianPowerCount)
+{
+    size_t count = 0;
+    for (const auto& it : cp)
+        count++;
```

Comment:
I'm not sure that `size_t count = std::accumulate(cp.cbegin(), cp.cend(), 0, [](size_t acc) {return acc + 1;});` would be simpler here?